src/libs/elektra/keyvalue.c

```diff
@@ -480,6 +493,11 @@ ssize_t keyGetBinary (const Key * key, void * returnedBinary, size_t maxSize)
  * won't be changed. This will only happen in the successful case,
  * but not when -1 is returned.
  *
+ * @pre @p newBinary is not NULL and @p dataSize > 0
+ * @pre @p key is not read-only
+ * @post @p key's value set exactly to the data in @p newBinary
```

Comment:
What if dataSize does not match the size of newBinary? I think we need one more precondition.